### ISSUE TITLE
[a11y]: Actionable Notification does not trap focus

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19302]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

n/a

### Description

Focus can be moved outside of Actionable Notification. From the storybook docs:

"This component grabs and traps focus until the action is acted upon or the notification is dismissed."

I think this may lead to potential WCAG 2.4.11 failures if the notification obscures another element on the page that is brought into focus without the notification being closed.

https://github.com/user-attachments/assets/959ddcff-2876-44b5-90b6-f9dd326177d7

### WCAG 2.1 Violation

2.4.11 (Focus not obscured)

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-notifications-actionable--default

### Steps to reproduce

1. Go to https://web-components.carbondesignsystem.com/?path=/story/components-notifications-actionable--default
2. Press TAB on keyboard until the action or close buttons of the notification come into focus (I recommend clicking "skip to canvas" to speed this up)
3. Continue pressing TAB and observe that focus can move outside of Notification without it closing

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
